src/PenSkin.js

```diff
@@ -218,7 +149,7 @@ class PenSkin extends Skin {
     drawLine (penAttributes, x0, y0, x1, y1) {
         // For compatibility with Scratch 2.0, offset pen lines of width 1 and 3 so they're pixel-aligned.
         // See https://github.com/LLK/scratch-render/pull/314
-        const diameter = penAttributes.diameter || DefaultPenAttributes.diameter;
```

Comment:
It looks like at https://github.com/LLK/scratch-vm/blob/9329f93e74ffbb924b12f318a41db69e1387db74/src/extensions/scratch3_pen/index.js#L188 we expect penState to have a penAttributes attribute, otherwise it will crash, which isn't guaranteed when pen state is set

It also seems like when pen state is set, we don't guarantee that penAttributes properly has the .diameter and .color4f attributes, which I guess is why DefaultPenAttributes existed. Seems like where customState is set, we should be verifying that the format of the object is correct in order to safely remove this safety check

Reply:
I'll just leave out the default pen attributes changes for now until that can be revisited in the VM